pkg/compactor/compactor.go

```diff
@@ -36,21 +36,24 @@ import (
 )
 
 const (
-	blocksMarkedForDeletionName = "cortex_compactor_blocks_marked_for_deletion_total"
-	blocksMarkedForDeletionHelp = "Total number of blocks marked for deletion in compactor."
+	blocksMarkedForDeletionName  = "cortex_compactor_blocks_marked_for_deletion_total"
+	blocksMarkedForDeletionHelp  = "Total number of blocks marked for deletion in compactor."
+	blocksMarkedForNoCompactName = "cortex_compactor_blocks_marked_for_no_compact_total"
+	blocksMakredForNoCompactHelp = "Total number of blocks marked for no compact in compactor"
 )
 
 var (
 	errInvalidBlockRanges = "compactor block range periods should be divisible by the previous one, but %s is not divisible by %s"
 	RingOp                = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, nil)
 
-	DefaultBlocksGrouperFactory = func(ctx context.Context, cfg Config, bkt objstore.Bucket, logger log.Logger, reg prometheus.Registerer, blocksMarkedForDeletion prometheus.Counter, garbageCollectedBlocks prometheus.Counter) compact.Grouper {
+	DefaultBlocksGrouperFactory = func(ctx context.Context, cfg Config, bkt objstore.Bucket, logger log.Logger, reg prometheus.Registerer, blocksMarkedForNoCompact prometheus.Counter, blocksMarkedForDeletion prometheus.Counter, garbageCollectedBlocks prometheus.Counter) compact.Grouper {
 		return compact.NewDefaultGrouper(
 			logger,
 			bkt,
 			false, // Do not accept malformed indexes
 			true,  // Enable vertical compaction
 			reg,
+			blocksMarkedForNoCompact,
```

Comment:
The order of arguments is wrong here. `blocksMarkedForNoCompact` is not in this position.